File: WebApplication/src/modules/report/models/report-search-result.dto.ts
import { DamageDto } from "./damage.dto";
import { ReportedDamageDto } from "./reported-damage.dto";
import { ThreatDto } from "./threat.dto";

export class ReportSearchResultDto {
  itemName: string;
  itemCount: number;
  threat: ThreatDto;
  damage: DamageDto;
  reportedDamage: ReportedDamageDto;
}

File: Zopoesht.Application/Reports/Dtos/ReportSearchResultDto.cs
namespace Zopoesht.Application.Reports.Dtos
{
    public class ReportSearchResultDto
    {
        public ReportSearchResultDto(string itemName)
        {
            ItemName = itemName;
            Threat = new ThreatDto();
            Damage = new DamageDto();
            ReportedDamage = new ReportedDamageDto();
        }

        public string ItemName { get; set; }

        public int ItemCount { get; set; }

        public ThreatDto Threat { get; set; }

        public DamageDto Damage { get; set; }

        public ReportedDamageDto ReportedDamage { get; set; }
    }
}
